gst/tensor_filter/tensor_filter.c

```diff
@@ -853,6 +853,62 @@ gst_tensor_filter_load_tensor_info (GstTensorFilter * self)
   }
 }
 
+/**
+ * @brief Printout the comparison results of two tensors.
+ * @param[in] info1 The tensors to be shown on the left hand side
+ * @param[in] info2 The tensors to be shown on the right hand side
+ * @return The newly allocated string for the printout. The caller must deallocate it.
+ * @todo If this is going to be used by other elements, move this to nnstreamer/tensor_common.
+ */
+static gchar *
+_compare_tensors (GstTensorsInfo * info1, GstTensorsInfo * info2)
+{
+  gchar null[] = "";
+  gchar *result = null;
+  int i;
+
+  for (i = 0; i < NNS_TENSOR_SIZE_LIMIT; i++) {
+    gchar *line, *tmp;
+    gchar *left = NULL, *right = NULL;
+
+    if (info1->num_tensors <= i && info2->num_tensors <= i)
+      break;
+
+    if (info1->num_tensors > i)
+      left = g_strdup_printf ("%s [%u:%u:%u:%u]",
```

Comment:
dependency of tensor rank limit (info1 and info2)
IMO, it is better to use get_tensor_dimension_string() instead.
dim_str = get_tensor_dimension_string (info->info[i].dimension);
left = g_strdup_printf ("%s [%s]", ...);
g_free (dim_str);

Reply:
Applied. Thianks!